src/lib.rs

```diff
@@ -83,6 +90,8 @@ impl MultiFernet {
     }
 }
 
+type ParsedToken = (Cursor<Vec<u8>>, Vec<u8>, Vec<u8>);
```

Comment:
Made this into a struct with is a bit easier to document. I attempted to avoid using owned fields (specifically for ciphertext), but couldn't get the lifetimes to work out (there's a lot of overlapping references), but at least the secrets I've seen are fairly small so it shouldn't be _too_ bad.